src/index.js

```diff
@@ -19,7 +19,8 @@ import rules from "./build/rules.js";
 /** @typedef {import("eslint").Linter.RulesRecord} RulesRecord*/
 /** @typedef {import("eslint").Linter.Config} Config*/
 /** @typedef {import("eslint").ESLint.Plugin} Plugin */
-/** @typedef {import("eslint").Rule.RuleModule} RuleModule */
+/** @typedef {import("./types.ts").MarkdownRuleDefinition} RuleModule */
```

Comment:
`RuleModule` is now an alias for the new `MarkdownRuleDefinition` type with the default generic parameters. It's still there for compatibility but I think it's now confusing (it's no longer identical to ESLint's `RuleModule`). So I'm not sure if we should also export `MarkdownRuleDefinition` as a generic type.

Reply:
Yeah, I'd say just use `MarkdownRuleDefinition`. People can always switch to ESLint's version of `RuleModule` if they really need it.

Reply:
I've added an export for `MarkdownRuleDefinition` as a generic type in 896a1949a16731abfa165e315ab5b19ce1b8772b. Now this can be used to define a custom Markdown rule with `RuleOptions`, `MessageIds` and `ExtRuleDocs`, all optional properties.

Reply:
Note: we need to discuss the implementation details for `MarkdownRuleDefinition` along with `JSRuleDefinition` in eslint/eslint#19521.